servant-server/src/Servant/Server/Internal.hs

```diff
@@ -351,12 +361,23 @@ instance (KnownSymbol sym, FromHttpApiData a, HasServer api context)
         -- named "foo" or "foo[]" and call parseQueryParam on the
         -- corresponding values
         parameters r = filter looksLikeParam (querytext r)
-        values r = mapMaybe (convert . snd) (parameters r)
-    in  route (Proxy :: Proxy api) context (passToServer subserver values)
+        parseParam (paramName, paramTxt) =
+          case parseQueryParam (fromMaybe "" paramTxt) of
```

Comment:
Hmm, I guess we have to decide what to when

```hs
type API = QueryParams "foo" MyType :> ...
```

and querystring is

```
foo&foo&foo[]=1&foo[]=2
```

should we pass an empty string, or strip them?

based on

```diff
-          convert Nothing = Nothing
 -          convert (Just v) = parseQueryParamMaybe v
```

we did strip empty params